python/lsst/pipe/base/graphBuilder.py

```diff
@@ -508,6 +535,12 @@ def connectDataIds(self, registry, collections, userQuery, externalDataId):
             results, just as if these constraints had been included via ``AND``
             in ``userQuery``.  This includes (at least) any instrument named
             in the pipeline definition.
+        datasetQueryQconstraint : `str`
```

Comment:
How hard would it be to make this accept a list of `str`, and do the string splitting in `ctrl_mpexec`, with enum sentinals or something for `all` and `off`?  (Oh, if we only had stateful enums...)

Reply:
I figured the work had to happen somewhere and putting the logic here centralized it from potentially other places potentially doing it repeatedly, but I dont have a really strong feeling, I don't like it anywhere, and really I dont have the kind of data structures I wanted to use anyway.... If I have time I will move it if you want

Reply:
I think the logic that converts from a single `str` is really just for command-line interfaces, and is fine to leave in ctrl_mpexec (or maybe the CLI utilities in daf_butler); any Python interface (even a JSON interface someday) should really be using something more structured.

But I could also easily believe that adding something that works to the CLI utilities is actually a lot harder, if only because it requires loading Click stuff into brain-memory from cold storage, so not a priority.